pegasus_wrapper/docker_job_script.py

```diff
@@ -0,0 +1,201 @@
+        return DockerJobScriptGenerator(
+            spack_config=SpackConfiguration.from_parameters(params)
+        )
+
+    def write_shell_script_to(
+        self,
+        docker_image_name: str,
+        docker_command: str,
+        docker_tar_path: str,
+        *,
+        working_directory: Path,
+        script_path: Path,
+        cmd_args: str = None,
+        ckpt_path: Optional[Path] = None,
+        pre_job: str = "",
+        post_job: str = "",
+    ) -> None:
+        script_path.write_text(
+            self.generate_shell_script(
+                docker_image_name=docker_image_name,
+                docker_command=docker_command,
+                docker_tar=docker_tar_path,
+                cmd_args=cmd_args,
+                working_directory=working_directory,
+                ckpt_path=ckpt_path,
+                pre_job=pre_job,
+                post_job=post_job,
+            ),
+            encoding="utf-8",
+        )
+        # Mark the generated script as executable.
+        script_path.chmod(script_path.stat().st_mode | stat.S_IEXEC)
+
+    def write_service_shell_script_to(
+        self,
+        docker_image_name: str,
+        docker_image_path: str,
+        docker_args: str,
+        *,
+        start_script_path: Path,
+        stop_script_path: Path,
+        remove_docker_on_exit: bool = True,
+    ) -> None:
+
+        start_script_path.write_text(
+            self.start_docker_script_text(
+                docker_name=docker_image_name,
+                docker_args=docker_args,
+                docker_img=docker_image_path,
+                remove_on_exit=remove_docker_on_exit,
+            ),
+            encoding="utf-8",
+        )
+
+        stop_script_path.write_text(
+            self.stop_docker_script_text(docker_name=docker_image_name)
+        )
+        # Mark the generated script as executable.
+        start_script_path.chmod(start_script_path.stat().st_mode | stat.S_IEXEC)
+        stop_script_path.chmod(stop_script_path.stat().st_mode | stat.S_IEXEC)
+
+    def generate_shell_script(
+        self,
+        docker_image_name: str,
+        docker_command: str,
+        docker_tar: str,
+        *,
+        cmd_args: str = "",
+        working_directory: Path,
+        ckpt_path: Optional[Path] = None,
+        pre_job: str = "",
+        post_job: str = "",
+    ) -> str:
+        """
+        Returns the content of a shell script to run a container from the
+        given Docker image according to params
+        If `ckpt_path` is provided this script will generate a `check point` to allow
+        for future runs of the DAX to skip already generated outputs. This can be useful
+        for outputs which are time or resource intensive or can be shared across projects.
+        Examples: Running BERT on all of Gigaword or Converting RDF triples to FlexNLP documents
+        """
+        docker_job = DOCKER_JOB.format(
+            docker_image=f"{docker_image_name}",
```

Comment:
Ok, this may be my lack of robust understanding of docker. @danielnapierski To confirm my understanding is correct as I address variable names:
Docker Image Name: The result of a docker build and what is called to run a container
Docker Container Name: A specific running instance of a docker image. (Set a docker container name via the `--name` argument?)

Reply:
I've updated the variable name to `docker_container_name`